backends/events/androidsdl/androidsdl-events.cpp

```diff
@@ -43,16 +42,9 @@ bool AndroidSdlEventSource::handleMouseButtonDown(SDL_Event &ev, Common::Event &
 	else if (ev.button.button == SDL_BUTTON_MIDDLE) {
 		event.type = Common::EVENT_MBUTTONDOWN;
 
-		static int show_onscreen = 0;
-		if (show_onscreen == 0) {
-			SDL_ANDROID_SetScreenKeyboardShown(0);
-			show_onscreen++;
-		} else if (show_onscreen==1) {
-			SDL_ANDROID_SetScreenKeyboardShown(1);
-			show_onscreen++;
-		}
-		if (show_onscreen == 2)
-			show_onscreen = 0;
+		static bool show_onscreen = g_system->getFeatureState(OSystem::kFeatureOnScreenControl);
```

Comment:
static - not neccesarily here.
No needs in const, because this variable changes value in next line.